### ISSUE TITLE
create various types for exporting with the write_scdf function

### ISSUE THREAD

**Issue by jazznbass** [jazznbass/scan#77]
write_scdf only exports to csv files while read_scdf imports csv, xlsx, and yml files.

write_scdf could also export xlsx and yml files.

**Comment by researchtool** [comment]
I like the idea! xlsx-Export (i.e. workaround to import csv into Excel) is a question/request I answer regularly.

**Comment by jazznbass** [comment]
Although

```r
whatever_export_function(as.data.frame(scdf), "my_filename.ext")

# e.g. 
# openxlsx::write.xlsx(as.data.frame(scdf), "my_filename.xlsx")

```
works. And we would not have to destabilize our package by importing the complex openxlsx, or readODS.
Maybe, a good description in the scan-book and the help files is better?

**Comment by researchtool** [comment]
fair point. Implementing a native xlsx-Export could also be time-consuming in the long run (you never know what they'll do their file formats and would always have to reverse engineer) ;)

**Comment by jazznbass** [comment]
I will tidy up the corresponding read and write functions:

as_scdf() convert a data frame into an scdf.
as.data.frame() converts an scdf into a data.frame.

(as.data.frame is a method for the as.data.frame generic function. That is why is has those dots in the function name)

write_scdf is already just a wrapper that calls as.data.frame and the write.csv function.

and read_scdf become just two wrappers that calls read.table / read_xlsx and calls as_scdf().

(with the exception of reading yml files ... but this is just a personal favorite that I am still developing)

**Comment by jazznbass** [comment]
implemented in branch 0.55.x
